context.go

```diff
@@ -9,24 +9,26 @@ const (
 	requestLimitKey
 )
 
+const _NoLimit = -1
+
 func WithIncrement(ctx context.Context, value int) context.Context {
 	return context.WithValue(ctx, incrementKey, value)
 }
 
-func getIncrement(ctx context.Context) int {
-	if value, ok := ctx.Value(incrementKey).(int); ok {
-		return value
-	}
-	return 1
+func getIncrement(ctx context.Context) (int, bool) {
+	value, ok := ctx.Value(incrementKey).(int)
+	return value, ok
+}
```

Comment:
on no ctx key, we should by default return 1

Reply:
I do it in the only place where this is used: https://github.com/klaidliadon/httprate/blob/ac748ea5a6b8921aef7a442985c77657a989932f/limiter.go#L90-L97
```go
	increment, ok := getIncrement(r.Context())
	if !ok {
		increment = 1
	}
	// If the increment is 0, we do not limit
	if increment == 0 {
		return false
	}
```